packages/client/src/LensClient.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { Wallet } from 'ethers';
+
+import LensClient, { Credentials, mumbai, ReactionTypes } from '.';
+
+let credentials: Credentials;
+const getCredentials = async (wallet: Wallet) => {
+  if (credentials) {
+    return credentials;
+  }
+  const client = LensClient.init(mumbai);
+  const address = await wallet.getAddress();
+  const challenge = await client.generateChallenge(address);
+  const signature = await wallet.signMessage(challenge);
+
+  credentials = await client.authenticate(address, signature);
+  return credentials;
+};
+
+const getAuthenticatedClient = async (wallet: Wallet) => {
+  const client = LensClient.init(mumbai);
+  const address = await wallet.getAddress();
+  const challenge = await client.generateChallenge(address);
+  const signature = await wallet.signMessage(challenge);
+
+  await client.authenticate(address, signature);
+  return client;
+};
+
+describe(`Given a LensClient configured for testnet and a wallet`, () => {
+  const wallet = new Wallet(
+    'ca9a3a3d4026e6228713e683a9c45ef65a538b2f9336813bd597f5effa38668d', // found on reddit
+  );
+
+  describe(`when authenticating with wallet signature`, () => {
+    it(`authenticates with success`, async () => {
+      const { accessToken, refreshToken } = await getCredentials(wallet);
+
+      expect(accessToken.length).toBeGreaterThan(1);
+      expect(refreshToken.length).toBeGreaterThan(1);
+    });
+  });
+
+  describe(`when refreshing authentication from stored credentials`, () => {
+    it(`authenticates with success`, async () => {
+      const { refreshToken: storedRefreshToken } = await getCredentials(wallet);
+      const client = LensClient.init(mumbai);
+
+      const { accessToken, refreshToken } = await client.refreshCredentials(storedRefreshToken);
+
+      expect(accessToken.length).toBeGreaterThan(1);
+      expect(refreshToken.length).toBeGreaterThan(1);
+    });
+  });
+
+  describe(`when checking if access token is valid`, () => {
+    it(`returns true`, async () => {
```

Comment:
Even after adopting a gherkin-line syntax, tests written like this one fails to document the desired behaviour:
```
Given a LensClient configured for testnet and a wallet
when checking if access token is valid
[it] should return true
```

If we are to treat the test description as a specification, the specification above would lead to an odd result.

```
isAccessTokenValid(_: string) {
  return true;
}
```

Probably the intent of this specific test is not about the actual underlying `isAccessTokenValid` details, that seems more appropriate form `Authentication.spec.ts`, is the intent here to be an integration test of some sort? Maybe we can try communicate that.

Reply:
this is testing if `isAccessTokenValid` is correctly exposed directly from LensClient, I will improve test description